src/instructlab/lab.py

```diff
@@ -667,6 +668,50 @@ def chat(
         if not api_base:
             api_base = ctx.obj.config.serve.api_base()
 
+    # if only the chat is running (`ilab chat`) and the temp server is not, the chat interacts
+    # in server mode (`ilab serve` is running somewhere) and thus we want to print the model used by
+    # the server
+    if not is_temp_server_running():
+        # If the model from the CLI and the one in the config are the same, use the one from the
+        # server if they are different else let's use what the user provided
+        #
+        # 'model' will always get a value and never be None so it's hard to distinguish whether
+        # the value came from the user input or the default value.
+        # We can only assume that if the value is the same as the default value and the value
+        # from the config is the same as the default value, then the user didn't provide a value
+        # we then compare it with the value from the server to see if it's different
+        if (
+            model == config.DEFAULT_MODEL
+            and ctx.obj.config.chat.model == config.DEFAULT_MODEL
```

Comment:
I think this condition needs to include verifying that the API endpoint URL is the same as the one configured for `ilab serve`. We don't want this logic running if someone did `ilab chat --endpoint-url ${SOMETHING_REMOTE}

Reply:
i fixed this